src/analysis/calculus/cont_diff.lean

```diff
@@ -576,36 +591,27 @@ begin
           rw [snoc_last, init_snoc] } } } }
 end
 
-/-- One direction of `cont_diff_within_at_succ_iff_has_fderiv_within_at`, but where all derivatives
-  are taken within the same set. -/
-lemma cont_diff_within_at.has_fderiv_within_at_nhds {n : ℕ}
-  (hf : cont_diff_within_at 𝕜 (n + 1 : ℕ) f s x) :
-  ∃ u ∈ 𝓝[insert x s] x, u ⊆ insert x s ∧ ∃ f' : E → E →L[𝕜] F,
-    (∀ x ∈ u, has_fderiv_within_at f (f' x) s x) ∧ cont_diff_within_at 𝕜 n f' s x :=
-begin
-  obtain ⟨u, hu, f', huf', hf'⟩ := cont_diff_within_at_succ_iff_has_fderiv_within_at.mp hf,
-  obtain ⟨w, hw, hxw, hwu⟩ := mem_nhds_within.mp hu,
-  rw [inter_comm] at hwu,
-  refine ⟨insert x s ∩ w, inter_mem_nhds_within _ (hw.mem_nhds hxw), inter_subset_left _ _,
-    f', λ y hy, _, _⟩,
-  { refine ((huf' y $ hwu hy).mono hwu).mono_of_mem _,
-    refine mem_of_superset _ (inter_subset_inter_left _ (subset_insert _ _)),
-    refine inter_mem_nhds_within _ (hw.mem_nhds hy.2) },
-  { exact hf'.mono_of_mem (nhds_within_mono _ (subset_insert _ _) hu) }
-end
-
 /-- A version of `cont_diff_within_at_succ_iff_has_fderiv_within_at` where all derivatives
-  are taken within the same set. This lemma assumes `x ∈ s`. -/
-lemma cont_diff_within_at_succ_iff_has_fderiv_within_at_of_mem {n : ℕ} (hx : x ∈ s) :
+  are taken within the same set. -/
+lemma cont_diff_within_at_succ_iff_has_fderiv_within_at' {n : ℕ} :
```

Comment:
I am not completely convinced that this lemma is strictly stronger than the previous one: in the right to left direction, the previous one had weaker assumptions since `u` was allowed to be smaller. Could you restore some version of the old lemma (hopefully this should be trivial with the API you have developed)?

Reply:
What previous lemma do you mean? If you mean the `cont_diff_within_at_succ_iff_has_fderiv_within_at_of_mem` that I removed, notice that that lemma had the assumption `x ∈ s` so `insert x s = s`

Reply:
Ah yes, thanks for the clarification!